CI: jedi v0.18.0 breaks a number of ipython test_tab_complete_warning tests

It looks like jedi version 0.18.0 was released yesterday and broke the following tests:

FAILED pandas/tests/arrays/categorical/test_warnings.py::TestCategoricalWarnings::test_tab_complete_warning
FAILED pandas/tests/frame/test_api.py::TestDataFrameMisc::test_tab_complete_warning[DataFrame]
FAILED pandas/tests/frame/test_api.py::TestDataFrameMisc::test_tab_complete_warning[Series]
FAILED pandas/tests/indexes/test_base.py::TestIndex::test_tab_complete_warning
FAILED pandas/tests/resample/test_resampler_grouper.py::test_tab_complete_ipython6_warning

in the Linux py38_Locale pipeline only.

I have looked into it and it seems that it is caused by this commit which is part of version 0.18.0 of Jedi
https://github.com/davidhalter/jedi/commit/abb2250bf5b11cd0b0733949311df54ca0adbbf1

Ipython has a commit that I believe will fix the error:
https://github.com/ipython/ipython/commit/dcd9dc90aee7e4c5c52ce44c18e7518934790612
but it's slated for inclusion in version 8.0 of ipython:
https://github.com/ipython/ipython/issues/11816
(version 7.19.0 is the most recent release, on 2020-10-30)

It's causing failures in all PRs that were submitted today (2020-12-26) starting with #38696 

- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

--- a/ci/deps/actions-38-locale.yaml
+++ b/ci/deps/actions-38-locale.yaml
@@ -18,7 +18,7 @@ dependencies:
   - html5lib
   - ipython
   - jinja2
-  - jedi<0.18.0
+  - jedi
   - lxml
   - matplotlib<3.3.0
   - moto

